dts/bindings/arm/nxp,bootpin.yaml

```diff
@@ -0,0 +1,17 @@
+# Copyright 2024, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  NXP Boot pins binding. This binding does not represent a full driver, but
+  declares pinctrl properties, so that the user can configure boot pins
+  at the board level.
```

Comment:
Remember that DT describes hardware, so comments about system-level stuff (driver, pinctrl, etc.) should be avoided. Just describe What this binding is for, and maybe a usage example.